pmoves/tools/chit_encode_hook.py

```
@@ -0,0 +1,290 @@
+    text: str,
+    depth: float = 0.5,
+    semantic_angle: Optional[float] = None,
+) -> HyperbolicCoords:
+    """Map content to Poincare disk coordinates.
+
+    Hierarchical depth maps to radial distance (deeper = closer to boundary).
+    Semantic angle determined by content hash for consistent positioning.
+    """
+    # Hash-based angle if not provided
+    if semantic_angle is None:
+        text_hash = hashlib.md5(text[:200].encode("utf-8")).hexdigest()
```

Ruff (0.15.1)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)